hq/public/javascripts/app.js

```diff
@@ -7,68 +7,80 @@ jQuery(function($) {
   }
 });
 
-$(document).ready(function() {
+$(document).ready(function () {
   // initalizing the mobile nav and the modal
   $('.button-collapse').sideNav();
   $('.modal').modal();
   // Make any tables with the filter class filterable
   $('.filterable-table').filterTable();
 
-  $('.js-iam-expand').click(function() {
+  $('.js-iam-expand').click(function () {
     $('.collapsible-header').addClass('active');
     $('.collapsible-body').css('display', 'block');
   });
 
-  $('.js-iam-collapse').click(function() {
-    $('.collapsible-header').removeClass(function() {
+  $('.js-iam-collapse').click(function () {
+    $('.collapsible-header').removeClass(function () {
       return 'active';
     });
     $('.collapsible-body').css('display', 'none');
   });
 
   // Extra interactions for the dropdowns on the Security Groups page
   $('.js-finding-details').hover(
-    function() {
+    function () {
       $(this).collapsible('open', 0);
     },
-    function() {
+    function () {
       $(this).collapsible('close', 0);
     }
   );
 
   // filtering table results
-  $('.js-finding-filter').change(function() {
-    $('#show-ignored-findings')[0].checked
-      ? $('.finding-suppressed--true').show()
-      : $('.finding-suppressed--true').hide();
-    $('#show-flagged-findings')[0].checked
-      ? $('.finding-suppressed--false').show()
-      : $('.finding-suppressed--false').hide();
-  });
-  $('.js-finding-filter-for-s3').change(function() {
-    $('#show-unencrypted-findings')[0].checked
-      ? $('.finding-unencrypted').show()
-      : $('.finding-unencrypted').hide();
-  });
-  $('.js-finding-filter-for-gcp').change(function() {
-    $('#show-critical-findings')[0].checked
-        ? $('.finding-critical').show()
-        : $('.finding-critical').hide();
-    $('#show-high-findings')[0].checked
-        ? $('.finding-high').show()
-        : $('.finding-high').hide();
-    $('#show-medium-findings')[0].checked
-        ? $('.finding-medium').show()
-        : $('.finding-medium').hide();
-    $('#show-low-findings')[0].checked
-        ? $('.finding-low').show()
-        : $('.finding-low').hide();
-    $('#show-unknown-findings')[0].checked
-        ? $('.finding-unknown').show()
-        : $('.finding-unknown').hide();
-  });
+  const form = document.querySelector('form.finding-filter');
+  if (form) {
+    const ignoredFindings = $('.finding-suppressed--true');
+    const flaggedFindings = $('.finding-suppressed--false');
+    const unencryptedFindings = $('.finding-unencrypted');
+    const criticalFindings = $('.finding-critical');
+    const highFindings = $('.finding-high');
+    const mediumFindings = $('.finding-medium');
+    const lowFindings = $('.finding-low');
+    const unknownFindings = $('.finding-unknown');
+
+    form.addEventListener('input', function () {
+      const formData = new FormData(form);
+
+      const {
```

Comment:
[Destructure](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment) the form; the field names map to the `name` attribute on the `input` element in the form.